sampledkd/training/entrypoint_utils.py

```
@@ -0,0 +1,432 @@
+    if tok.pad_token_id is None and tok.eos_token is not None:
+        tok.pad_token = tok.eos_token
+
+    # Filter teacher GPUs (exclude student's GPU if present)
+    teacher_gpus = [g for g in prefer_gpus if g != student_gpu]
+    if not teacher_gpus:
+        raise RuntimeError("No GPUs available for teacher after excluding the student GPU.")
+
+    # ===== 1) 8-bit quantization on single GPU =====
+    try:
+        if not _bnb_triton_available():
+            raise RuntimeError("bitsandbytes/triton not available; skipping 8-bit fallback")
```

Ruff (0.13.3)

Abstract `raise` to an inner function

(TRY301)